### ISSUE TITLE
4.0.10 crashed on Windows 7 caused by gdiplus

### ISSUE THREAD

**Issue by sinri** [shadowsocks/shadowsocks-windows#1922]
Version 4.0.9 runs well, while upgraded to 4.0.10, would crash soon on launch.
Windows 7, .NET Framework 4.7.2, and C++ Redistribution 2005, 2008, 2010, 2012, 2013, 2017 installed.
Try to install C++ Redistribution 2015, but showed conflict with other version.

Checked Windows Event Viewer and found following.

```
应用程序: Shadowsocks.exe
Framework 版本: v4.0.30319
说明: 由于未经处理的异常，进程终止。
异常信息: System.AccessViolationException
   在 System.Drawing.SafeNativeMethods+Gdip.GdipCreateFontFamilyFromName(System.String, System.Runtime.InteropServices.HandleRef, IntPtr ByRef)
   在 System.Drawing.FontFamily.CreateFontFamily(System.String, System.Drawing.Text.FontCollection)
   在 System.Drawing.FontFamily..ctor(System.String, Boolean)
   在 System.Drawing.Font.Initialize(System.String, Single, System.Drawing.FontStyle, System.Drawing.GraphicsUnit, Byte, Boolean)
   在 System.Drawing.Font..ctor(System.String, Single)
   在 Shadowsocks.Model.LogViewerConfig..ctor()
   在 Shadowsocks.Model.Configuration.Load()
   在 Shadowsocks.Controller.ShadowsocksController..ctor()
   在 Shadowsocks.Program.Main()
```

```
错误存储段 ，类型 0
事件名称: APPCRASH
响应: 不可用
Cab ID: 0

问题签名: 
P1: Shadowsocks.exe
P2: 4.0.10.0
P3: 5af468a6
P4: gdiplus.dll
P5: 6.1.7601.23894
P6: 59946055
P7: c0000005
P8: 000fd333
P9: 
P10: 

附加文件: 

可在此处获取这些文件: 
C:\Users\Administrator\AppData\Local\Microsoft\Windows\WER\ReportQueue\AppCrash_Shadowsocks.exe_18cb9645fe3ae48beb52be320773def2b1eea_10d114f6

分析符号: 
重新检查解决方案: 0
报告 Id: 57b48b1e-8411-11e8-8548-005056c00008
报告状态: 4
```

**Comment by chenshaoju** [comment]
Did you use some fonts optimize?

Like MacType?

**Comment by sinri** [comment]
@chenshaoju 
I have no such memory as it is a new machine replaced recently in my company, but I cannot determine actually whether your assumption does with it. I am not familiar with the display optimization on fonts.
Well, does the fonts optimization lead to some characters which could be found in certain places? I might check again on my system following your further advise. THX.

**Comment by celeron533** [comment]
Perhaps the anti-virus software did some actions.

**Comment by sinri** [comment]
@celeron533 
I removed all known Security Software such as 360 serials the first minute I received the PC.
Only those with name of Microsoft are still running here. 
(I do not like them but company infrastructure department would install them before distribution. )

**Comment by celeron533** [comment]
Weird... it is very likely the system is corrupted.
Please also check the font settings in `gui-config.json`
```
  "logViewer": {
    "topMost": true,
    "wrapText": true,
    "toolbarShown": false,
    "Font": "Consolas, 8pt",
    "BackgroundColor": "Black",
    "TextColor": "White"
  },
```
and try to remove (or replace) the font.

**Comment by sinri** [comment]
@celeron533 Okay, I will try to check next week.

**Comment by sinri** [comment]
This issue is tested fixed on Version 4.1.1.
Directly use new version might be the solution.
